airbyte-webapp/src/components/Modal/Modal.module.scss

```diff
@@ -7,23 +7,39 @@
   }
 }
 
-.modal {
-  animation: fade-in 0.2s ease-out;
-  position: absolute;
+.backdrop {
+  position: fixed;
   top: 0;
   left: 0;
-  width: 100vw;
-  height: 100vh;
+  right: 0;
+  bottom: 0;
   background: rgba(colors.$black, 0.5);
+}
+
+.modalPageContainer {
+  z-index: 50;
```

Comment:
We have a central file now to manage z-indices (`_z_indices.scss`), so we can have this global values in one central place to make sure they are in the right order. Maybe we could move this to this file as well. I also feel we could go as high as 5000 or so here, to have enough space to both sides.